src/graph.cpp

```diff
@@ -135,17 +145,67 @@ void Graph::compute_distance_matrix() const {
   }
 }
 
-void Graph::add_node(boost::filesystem::path const &input_file,
+boost::filesystem::path Graph::add_node(boost::filesystem::path const &input_file,
                      Hash const &input_hash) {
-  assert(not has_node(input_file));
-#pragma omp critical(graph_)
-  {
+  if(not has_node(input_file)){
+//#pragma omp critical(graph_)
+  //{
+    if (passed_path_.find(input_file.parent_path()) == passed_path_.end()){ //path not parsed yet
+	passed_path_[input_file.parent_path()] = 1;
+	//std::cout << "YOP\n";
+    }
+
+
+    boost::filesystem::path no_path1 = "/.";
+    boost::filesystem::path no_path2 = ".";
+    std::string file = input_file.string();
+
+    if(input_file.parent_path() == no_path1 || input_file.parent_path() == no_path2){
+	//std::cout<<"/. or .\n";
+
+	for (it_p_=passed_path_.begin(); it_p_!=passed_path_.end();it_p_++){
+		if(it_p_->first != no_path1 && 
+		   it_p_->first != no_path2 &&
+		   has_node(it_p_->first / input_file.filename())){
+			return (it_p_->first / input_file.filename());
```

Comment:
For example, without this, if we 're scanning /sub-dir/one.dll depending to two.dll in the chroot (/two.dll) it won't find it and save it as it isn't here (/./two.dll). With this, if we've parsed /two.dll, it will find it and continues with this path.

Actually this function do this schema : 
/./dep.dll (not in the path) => do we have parsed it in an other directory?
   yes => save /the-dir/dep.dll
   no=> save /./dep.dll

/dir/dep2.dll found : 
    /./dep2.dll already saved ?
       yes : changes to /dir/dep2.dll
       no : save /dir/dep2.dll